### PR TITLE
 Seperate out SimpleReport into it's own file + Refactor #18 

### PR DESCRIPTION
**Pull request by antoniaklja**
I've separated simpleReport from main.go and added saving report to file flag.

### CODE REVIEW

**Comment by s-rah** [comment]
Thanks! Reporting looks good - however, the headers need to be normalized - currently have an example where `X-XSS-Protection` but the check is failing because the site returns `X-Xss-Protection`

**Comment by antoniaklja** [comment]
Sure, I will prepare it soon.

**Comment by antoniaklja** [comment]
Done, please merge if everything is fine.

File: report/report_generator.go [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+	if report.ResponseHeaders != nil {
+		if _, ok := report.ResponseHeaders["X-FRAME-OPTIONS"]; !ok {
+			buffer.WriteString("\033[091mLow Risk:\033[0m missing X-Frame-Options HTTP header discovered!\n")
+			buffer.WriteString("\t Why this is bad: Provides Clickjacking protection. Values: deny - no rendering within a frame, sameorigin - no rendering if origin mismatch, allow-from: DOMAIN - allow rendering if framed by frame loaded from DOMAIN\n")
+			buffer.WriteString("\t To fix, use X-Frame-Options: deny\n")
+		}
+		if _, ok := report.ResponseHeaders["X-XSS-PROTECTION"]; !ok {
+			buffer.WriteString("\033[091mLow Risk:\033[0m missing X-XSS-Protection HTTP header discovered!\n")
+			buffer.WriteString("\t Why this is bad: his header enables the Cross-site scripting (XSS) filter built into most recent web browsers. It's usually enabled by default anyway, so the role of this header is to re-enable the filter for this particular website if it was disabled by the user. This header is supported in IE 8+, and in Chrome (not sure which versions). The anti-XSS filter was added in Chrome 4. Its unknown if that version honored this header.\n")
```
Comment (s-rah) [inline report/report_generator.go]: [T]his header

Also, I think we should drop the lines about IE 8 and Chrome. (And does TBB respect this header)?

**Comment by antoniaklja** [comment]
Good idea, I will delete it. In order to your question TBB is fork of mozilla browser so I think that it is respected.

**Comment by s-rah** [comment]
Merged in 129ee4f8f4bdc34b826cf29cd9ae6a982921f5c4 - I fixed up some of the content, and downgraded the warnings. I don't think many of these headers apply to typical hidden service threat models (due to the reduce risk from lack of Javascript), I think we need to do more research to determine the right message to inform operators of what to do - but setting these headers is unlikely to be harmful and may actually help.

**Pull request closed without merging**